src/WebJobs.Script.WebHost/Configuration/ScriptApplicationHostOptionsSetup.cs

```diff
@@ -61,6 +67,44 @@ public void Configure(string name, ScriptApplicationHostOptions options)
                 options.IsSelfHost = options.IsSelfHost;
                 options.IsStandbyConfiguration = true;
             }
+
+            options.IsFileSystemReadOnly = IsZipDeployment(options);
+        }
+
+        private bool BlobExists()
+        {
+            return _cloudBlockBlobHelperService.BlobExists(_environment.GetEnvironmentVariable(ScmRunFromPackage)).GetAwaiter().GetResult();
+        }
+
+        private bool IsZipDeployment(ScriptApplicationHostOptions options)
+        {
+            // If the app is using app settings for run from package, we don't need to check further, it must be a zip deployment.
+            bool runFromPkgConfigured = IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteZipDeployment)) ||
```

Comment:
Can we avoid duplicating these settings here, perhaps by leveraging you refactored utility method, so we could call _environment.IsRunFromZipOrPackage() here so the logic is centralized?

Reply:
Before my changes, `_environment.IsZipDeployment(bool validate = true)` had two checks.
1) If `validate=true` then we utilize `IsValidZipSetting` to check the app settings.
2) If `validate=false` (this happens only in a single call in `WebJobsScriptHostService`) then we utilize `!string.IsNullOrEmpty` to check the app settings.

My change refactored 1) into this method in Options and 2) remained in environment. They are different checks, so there's no code duplication.

Can you please clarify what you are suggesting? I'm not sure I understand what you want to live in environment as opposed to here. Thanks :)

Reply:
I'm just trying to avoid having two disjointed blocks of code where we're maintaining the list of 3 setting we have to check. We centralized that in one place to ensure when we modify, it's all done in one place so people do it correctly. So I'm suggesting to retain the previous helper as it was, with any modifications you need to make for ScmRunFromPackage.

Reply:
> I'm just trying to avoid having two disjointed blocks of code where we're maintaining the list of 3 setting we have to check. 

Actually, there is a slight difference in the checks in the previous helper in Environment and the checks I added here. The difference is that, here, we omit checking for `!string.IsNullOrEmpty(environment.GetEnvironmentVariable(ScmRunFromPackage))`, so I can't use the previous helper here.